### PR TITLE
2.12: epic upgrade/simplification

### PR DESCRIPTION
**Pull request by SethTisue**
- no more spaces!!! one big space
- drop specs2 2.x (!!!), spray-twirl
- more jawn
- add scodec
- uncomment json4s
- update and/or unfork genjavadoc, scodec-bits, discipline,
  kind-projector, scalaz, scala-java8-compat
- update comments on status of various projects & settings
- remove commented-out ENSIME stuff, commented-out sbt/zinc stuff

### CODE REVIEW

**Comment by SethTisue** [comment]
https://scala-ci.typesafe.com/job/scala-2.12.x-integrate-community-build/726/consoleFull (404 til Jenkins catches up)

**Comment by SethTisue** [comment]
things were broken at various points along the way, but I've kept the separate commits to facilitate review, the combined diffs are hard to follow

**Comment by SethTisue** [comment]
screen shot 2016-10-04 at 10 04 11 pm

🎉

**Comment by SethTisue** [comment]
fyi @etorreborre @non @mpilquist @xuwei-k @eed3si9n @dwijnand @fommil @larsrh @BennyHill @retronym

**Comment by fommil** [comment]
I guess ensime is kind of blocked until we can get a 2.12 build going...

**Comment by ghost** [comment]
@SethTisue  w00t!!
